app/jinja_filters.py

```diff
@@ -79,11 +79,17 @@ def format_unit(
     # until removed from schema we substitute mass-tonne for mass-metric-ton before format unit
     measurement_unit = "mass-tonne" if unit == "mass-metric-ton" else unit
 
+    locale = (
+        DEFAULT_LOCALE
+        if str(flask_babel.get_locale()) in {"en", "eo"}
+        else flask_babel.get_locale()
+    )
+
     formatted_unit: str = units.format_unit(
         value=value,
         measurement_unit=measurement_unit,
         length=length,
-        locale=flask_babel.get_locale(),
+        locale=locale,
```

Comment:
Keep this as it was and update out `get_locale` method instead, then we set it correct in all places.

Reply:
Done, thanks!